superset/hive_query.py

```diff
@@ -39,72 +34,123 @@
     'P1Y'  : 31556926
 }
 
-# return seconds from gran valaue map
-def getGranValueInSeconds(value):
-    if  value in GRAN_VALUE_MAP:
+# return seconds from gran valaue map default is 86400
+def get_gran_value_in_seconds(value,time_partitions):
+    partition_min_grain = get_partitions_min_grain(time_partitions)
+    if value in GRAN_VALUE_MAP and  GRAN_VALUE_MAP[value] >= partition_min_grain:
         return GRAN_VALUE_MAP[value]
+    return partition_min_grain
+
+def get_partitions_min_grain(time_partitions):
+    partition_grains = list()
+    if 'year' in time_partitions:
+        partition_grains.append(31556926)
+    if 'month' in time_partitions:
+        partition_grains.append(2629743)
+    if 'day' in time_partitions:
+        partition_grains.append(86400)
+    if 'hour' in time_partitions:
+        partition_grains.append(3600)
+    if 'minute' in time_partitions:
+        partition_grains.append(60)
+
+    partition_grains.sort()
+
+    return partition_grains[0]
+
+
+
+def get_partitioned_query_format(time_partitions):
+    partition_seq = list()
+    if 'year' in time_partitions:
+        partition_seq.append(time_partitions['year']+" = %Y ")
+    if 'month' in time_partitions:
+        partition_seq.append(time_partitions['month']+" = %m ")
+    if 'day' in time_partitions:
+        partition_seq.append(time_partitions['day']+" = %d ")
+    if 'hour' in time_partitions:
+        partition_seq.append(time_partitions['hour']+" = %H ")
+    if 'minute' in time_partitions:
+        partition_seq.append(time_partitions['minute']+" = %M ")
+   
+    partition_seq_str = "AND ".join(partition_seq)
+    query_str  = "( "+ partition_seq_str+ ")"
+    return query_str
```

Comment:
Can we return None in case no partition is provided

Reply:
checks added prior to this function